tools/csw_cloudfront_distribution/cloudfront.tf

```
@@ -0,0 +1,73 @@
+# Strip protocol and trailing path from url to get domain
+locals {
+  api_gatway_url_components = "${split("/",var.api_gateway_url)}"
+  api_gateway_domain = "${join("",slice(local.api_gatway_url_components,2,3))}"
+  api_gateway_path = "/${join("/",compact(slice(local.api_gatway_url_components,3,length(local.api_gatway_url_components))))}"
+}
+
+resource "aws_cloudfront_distribution" "cf_distribution" {
+  depends_on = ["aws_route53_record.cert_validation_record"]
+  origin {
+    domain_name = "${local.api_gateway_domain}"
+    origin_id   = "${local.cf_origin_id}"
+    origin_path = "${local.api_gateway_path}"
+
+    custom_origin_config {
+      http_port = 80
+      https_port = 443
+      origin_protocol_policy = "https-only"
+      origin_ssl_protocols = ["TLSv1.2"]
+    }
+  }
+
+  enabled             = true
+  is_ipv6_enabled     = false
+  wait_for_deployment = false
+  comment             = "Managed by Terraform"
+
+  aliases = ["${local.target_url}"]
+
+  default_cache_behavior {
+    allowed_methods  = ["HEAD", "GET", "OPTIONS", "DELETE", "POST", "PUT", "PATCH"]
+    cached_methods   = ["GET", "HEAD"]
+    target_origin_id = "${local.cf_origin_id}"
+
+    forwarded_values {
+      query_string = true
+      headers      = ["CloudFront-Forwdared-Proto"]
```

Typo, I don't know how crucial this is but it might cause problems:
```suggestion
      headers      = ["CloudFront-Forwarded-Proto"]
```

---

Good spot. Fixed - I didn't need it in the end but it's probably good to keep it in case.